### ISSUE TITLE
In For Statement with 'index' key but no 'each', the 'index' is pouplated with values not indexes 

### ISSUE THREAD

**Issue by NikolayPeev** [AlexGilleran/jsx-control-statements#50]
I mean when I try this:
```
        <For index="index" of={[100, 200, 300]}>
          <span key={index}>{index}</span>
        </For>
```
I get 100, 200, 300 and I'd like to get 1, 2, 3.
As far as I saw, when 'each' is missing the index key is passed as first param, so it takes the place of 'each'.
Is this a known issue?

**Comment by AlexGilleran** [comment]
Huh. No not a known issue - I suppose it makes sense that it'd do that because if you don't specify an `each` attribute, `index` is the first parameter that gets added to the list of parameters for the `map` method. Thanks for reporting it!

**Comment by NikolayPeev** [comment]
So, is it a bug then?
I could add a simple check, and if there is no 'each', it could add a placeholder.

**Comment by AlexGilleran** [comment]
Yeah definitely a bug - the fix should be as simple as removing that assumption from the code, it shouldn't even need a specific check.

Sorry to leave this sitting, been trying to get a Random Hacks of Kindness project done and move house :(

**Comment by texttechne** [comment]
Fixed on current master, but not released yet.

**Comment by AlexGilleran** [comment]
Fixed in v3.1.3! :ok_hand:
